pep/app/pods/components/page/sidebar/widgets/read-later/component.ts

```diff
@@ -25,16 +32,28 @@ export default class PageSidebarWidgetsReadLater extends Component<PageSidebarWi
      *
      * @memberof PageSidebarWidgetsReadLater
      */
-    loadFromUserPreferences() {
-        this.results = this.currentUser.getDocuments(PreferenceKey.READ_LATER);
+    @restartableTask
+    *loadFromUserPreferences() {
+        const ids = this.currentUser.getDocuments(PreferenceKey.READ_LATER);
+        if (ids.length) {
+            const queryItems = ids.map((id) => {
+                return {
+                    id: SearchFacetId.ART_ID,
+                    value: id
+                };
+            });
+            const params = buildSearchQueryParams('', [], false, queryItems);
+            const results = yield this.store.query('document', params);
+            this.results = results.toArray();
+        }
     }
 
     /**
      * Load the widget results on render
      */
     @action
     onElementChange() {
```

Comment:
same here as the favorites component regarding renaming this method, e.g. `onPreferencesUpdate()`